### PR TITLE
Developed wait deploy step

### PR DESCRIPTION
**Pull request by nemo83**
Created a new step that waits for a given DeploymentId to complete and then updates the build accordingly.

I still have to update the README.md with the examples.

### CODE REVIEW

File: src/main/java/de/taimos/pipeline/aws/WaitDeployStep.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    @Extension
+    public static class DescriptorImpl extends AbstractStepDescriptorImpl {
+
+        public DescriptorImpl() {
+            super(Execution.class);
+        }
+
+        @Override
+        public String getFunctionName() {
+            return "waitDeploymentCompletion";
```
Comment (hoegertn) [inline src/main/java/de/taimos/pipeline/aws/WaitDeployStep.java]: maybe "awaitDeploymentCompletion" would be better or "waitForDeploymentCompletion"

File: src/main/java/de/taimos/pipeline/aws/WaitDeployStep.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        private static final String SucceededStatus = "Succeeded";
+
+        private static final String FailedStatus = "Failed";
+
+        @Override
+        protected Void run() throws Exception {
+            AmazonCodeDeployClient client = AWSClientFactory.create(AmazonCodeDeployClient.class, this.envVars);
+
+            String deploymentId = this.step.getDeploymentId();
+            Long maxWait = this.step.maxWait;
```
Comment (hoegertn) [inline src/main/java/de/taimos/pipeline/aws/WaitDeployStep.java]: please use getter here

File: src/main/java/de/taimos/pipeline/aws/WaitDeployStep.java [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    public static class Execution extends AbstractSynchronousNonBlockingStepExecution<Void> {
+
+        @Inject
+        private transient WaitDeployStep step;
+        @StepContextParameter
+        private transient EnvVars envVars;
+        @StepContextParameter
+        private transient TaskListener listener;
+
+        private static final Long PollingInterval = 10000L;
```
Comment (hoegertn) [inline src/main/java/de/taimos/pipeline/aws/WaitDeployStep.java]: Please use ALL_CAPS or start with lower-case letter for variables

Comment (nemo83) [inline src/main/java/de/taimos/pipeline/aws/WaitDeployStep.java]: Apologies, that's the scala convention. Fixing.

**Comment by nemo83** [comment]
Updated ChangeLog: added entry for `awaitDeploymentCompletion`.

I've also rebase master into branch so to avoid merge conflicts.

**Comment by hoegertn** [comment]
Thanks for the contribution. If you like, you can add step descriptions like S3 and withAWS by adding the files in the resources folder.

**Pull request merged**